### DISCUSSION TITLE
2-Factor Authentication Mystery

### DISCUSSION THREAD

**Discussion by tmeehan622** [community/community#133578] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

![authenticate](https://github.com/user-attachments/assets/f9389637-305d-410d-92c7-dea1312c3873)

I haven't logged into github for quite a while.  I've always logged in with a simple username and password. this time I get hit with the 2-factor authenticate screen.  (see attached screen shot) That's as far as I can get.  I'm familiar with 2-factor authentication in every other context ie I receive a 6-digit code via text or email - which i then use to log in.  That's clearly not how it's done with github. I'm dead in the water. BTW I have 2 github accounts.  The one I am posting this message from is not the one I am having trouble logging into. Can anyone explain how I should proceed to get past this road block.  Thank.

**Comment by InboraStudio** [comment]
Ah, yeaaaa, I totally get where you're coming from. GitHub's 2-factor authentication (2FA) can be kinda tricky if you haven't set it up or used it in a while. Here's the lowdown on what you need to do to get back into your account.

## So, basically, try these steps in order:

1 Check for backup codes.
2 Use your authentication app if you have one.
3 Look for any emails from GitHub.
4 Go through the recovery process if needed.
5 Contact GitHub Support if you're still stuck.

**Reply by tmeehan622** [reply]
Thanks for your reply.  When you say "Check for backup codes", where would one find backup codes.  It seems I would have had to generate such codes, which I have not.... Unless that's not where they would come from in which case where would these codes be located?  as far as authentication apps, I don't have one.  I did install an extension in chrome that somehow was generating codes that expired every 30 seconds or so.  They seem to be the wrong number of digits and when I tried one , it didn't work.  If it had works I would be wondering how that whole mechanism could even possibly work.  I began the recovery process but it think I even hit a wall with that - although I'm not sure how far I got with that.  Didn't really want to wait a few days to get into my account... but if it comes to that.... I guess i'll have to suck it up and go for it.  Thanks again for you reply.
